helix-core/src/test/java/org/apache/helix/integration/task/TestForceDeleteWorkflow.java

```diff
@@ -0,0 +1,433 @@
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed deleted for this
+    // workflow.
+    boolean isWorkflowDeleted = TestHelper.verify(() -> {
+      WorkflowConfig wfcfg = _driver.getWorkflowConfig(workflowName);
+      WorkflowContext wfctx = _driver.getWorkflowContext(workflowName);
+      IdealState is = _admin.getResourceIdealState(CLUSTER_NAME, workflowName);
+      return (wfcfg == null && wfctx == null && is == null);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowDeleted);
+
+    // Start the Controller
+    String controllerName = CONTROLLER_PREFIX + "_1";
+    _controller = new ClusterControllerManager(ZK_ADDR, CLUSTER_NAME, controllerName);
+    _controller.syncStart();
+  }
+
+  @Test(dependsOnMethods = "testDeleteCompletedWorkflowForcefully")
+  public void testDeleteRunningWorkflowForcefully() throws Exception {
+    // Create a simple workflow and wait until it reaches the Running state. Then delete the
+    // IdealState, WorkflowContext and WorkflowConfig.
+    String workflowName = TestHelper.getTestMethodName();
+    Workflow.Builder builder = createCustomWorkflow(workflowName, "100000", "0");
+    _driver.start(builder.build());
+
+    // Wait until workflow is created and goes to running stage.
+    boolean isWorkflowRunning = TestHelper.verify(() -> {
+      WorkflowContext wCtx1 = _driver.getWorkflowContext(workflowName);
+      return (wCtx1 != null && wCtx1.getWorkflowState() == TaskState.IN_PROGRESS);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowRunning);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed created for this
+    // workflow
+    Assert.assertNotNull(_driver.getWorkflowConfig(workflowName));
+    Assert.assertNotNull(_driver.getWorkflowContext(workflowName));
+    Assert.assertNotNull(_admin.getResourceIdealState(CLUSTER_NAME, workflowName));
+
+    // Stop the Controller
+    _controller.syncStop();
+
+    _driver.delete(workflowName, true);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed deleted for this
+    // workflow
+    boolean isWorkflowDeleted = TestHelper.verify(() -> {
+      WorkflowConfig wfcfg = _driver.getWorkflowConfig(workflowName);
+      WorkflowContext wfctx = _driver.getWorkflowContext(workflowName);
+      IdealState is = _admin.getResourceIdealState(CLUSTER_NAME, workflowName);
+      return (wfcfg == null && wfctx == null && is == null);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowDeleted);
+
+    // Start the Controller
+    String controllerName = CONTROLLER_PREFIX + "_2";
+    _controller = new ClusterControllerManager(ZK_ADDR, CLUSTER_NAME, controllerName);
+    _controller.syncStart();
+
+  }
+
+  @Test(dependsOnMethods = "testDeleteRunningWorkflowForcefully")
+  public void testDeleteStoppedWorkflowForcefully() throws Exception {
+    // Create a simple workflow. Stop the workflow and wait for completion. Delete the workflow forcefully afterwards
```

Comment:
Did you format this? If you didn't format it, you shouldn't check the box in the PR description.